### PR TITLE
[CIAS30-3698] leave only one controller to invitate participants to the intervention

### PR DESCRIPTION
**Pull request by WiktoriaRozanska**
## Related tasks
- [CIAS-3698](https://htdevelopers.atlassian.net/browse/CIAS30-3698)

## What's new?
- leave only one controller for inviting
- removed all duplicated code

### CODE REVIEW

**Review by OskarDybowskiHTD — COMMENTED** [review]
Nice job! 🔥

File: app/controllers/v1/interventions/invitations_controller.rb [hunk 1]
```diff
@@ -4,46 +4,56 @@
...
-      return render json: { message: I18n.t('interventions.invitations.wrong_intervention_type') }, status: :unprocessable_entity
-    end
-
     authorize! :create, Invitation
+    return head :not_acceptable unless intervention.published?
 
-    intervention.invite_by_email(intervention_invitation_params[:emails], intervention_invitation_params[:health_clinic_id])
-    render json: serialized_response(intervention_invitation_scope), status: :created
+    V1::Intervention::CreateInvitation.call(intervention, invitation_params[:invitations])
+    render json: serialized_response(invitation_scope), status: :created
```
[the remark this answers is not in the document]
Comment (WiktoriaRozanska) [reply app/controllers/v1/interventions/invitations_controller.rb:17]: Good point!

File: app/controllers/v1/interventions/invitations_controller.rb
[the remark this answers is not in the document]
Comment (WiktoriaRozanska) [reply app/controllers/v1/interventions/invitations_controller.rb:1]: I believe that leaving it in the "interventions" folder is as correct as possible. We have a lot of different invitations in the application and leaving it there immediately lets us know that it refers to an intervention or to a session in that intervention.

File: app/services/v1/intervention/create_invitation.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    object.invite_by_email(invitation_params[:emails], invitation_params[:health_clinic_id])
+  end
+
+  def fetch_object(invitation_params)
+    invitation_params['target_type'].classify.demodulize.safe_constantize.find(invitation_params['target_id'])
+  end
+
+  def wrong_type_of_invitation_for_object?(params, object)
+    target_type = params['target_type']
+    (target_type.eql?('intervention') && object.type.eql?('Intervention')) || (target_type.eql?('session') && object.intervention.type != 'Intervention')
```
[the remark this answers is not in the document]
Comment (WiktoriaRozanska) [reply app/services/v1/intervention/create_invitation.rb:38]: I think you didn't carefully read this condition. First, check if you try to invite to the intervention and the intervention has the wrong type and your suggestion works correctly `params['target_type'].camelize == object.type.deconstantize` => `'Intervention' == 'Intervention'` but the second condition check something different. If we get `session` we cannot have this check object is also a session. First of all, we have `Session::Classic` and `Session::CatMh` so your suggestion always fails. We can use `is_a?` but in this we always get true

**Review by OskarDybowskiHTD — APPROVED** [review]
🔥

**Pull request merged**